folium/plugins/timeline.py

```diff
@@ -0,0 +1,313 @@
+        **kwargs
+    ):
+        super().__init__(data)
+        self._name = "Timeline"
+
+        kwargs["point_to_layer"] = point_to_layer
+        kwargs["on_each_feature"] = on_each_feature
+        kwargs["style"] = style
+
+        if get_interval is not None:
+            kwargs["get_interval"] = get_interval
+
+        # extract JsCode objects
+        self.functions = {}
+        for key, value in list(kwargs.items()):
+            if isinstance(value, JsCode):
+                self.functions[camelize(key)] = value.js_code
+                kwargs.pop(key)
+
+        self.options = parse_options(**kwargs)
+
+    def render(self, **kwargs):
+        assert isinstance(
+            self._parent, Map
+        ), "Timeline can only be added to a Map object."
+        super().render(**kwargs)
+
+    def _get_self_bounds(self):
+        """
+        Computes the bounds of the object itself (not including it's children)
+        in the form [[lat_min, lon_min], [lat_max, lon_max]].
+
+        """
+        return get_bounds(self.data, lonlat=True)
+
+
+class TimelineSlider(JSCSSMixin, MacroElement):
+    """
+      Creates a timeline slider for timeline layers.
+
+      Parameters
+      ----------
+      auto_play: bool, default True
+          Whether the animation shall start automatically at startup.
+
+      start: str, int or float, default earliest 'start' in GeoJson
+          The beginning/minimum value of the timeline.
+      end: str, int or float, default latest 'end' in GeoJSON
+          The end/maximum value of the timeline.
+
+      enable_playback: bool, default True
+          Show playback controls (i.e. prev/play/pause/next).
+      enable_keyboard_controls: bool, default False
+          Allow playback to be controlled using the spacebar (play/pause) and
+          right/left arrow keys (next/previous).
+      show_ticks: bool, default True
+          Show tick marks on the slider
+      steps: int, default 1000
+          How many steps to break the timeline into.
+          Each step will then be (end-start) / steps. Only affects playback.
+      playback_duration: int, default 10000
+          Minimum time, in ms, for the playback to take. Will almost certainly
+          actually take at least a bit longer -- after each frame, the next
+          one displays in playback_duration/steps ms, so each frame really
+          takes frame processing time PLUS step time.
+
+    Examples
+    --------
+    See the documentation for Timeline
+
+    """
+
+    _template = Template(
+        """
+        {% macro header(this,kwargs) %}
+            <style>
+                .leaflet-bottom.leaflet-left {
+                    width: 100%;
+                }
+                .leaflet-control-container .leaflet-timeline-controls {
+                    box-sizing: border-box;
+                    width: 100%;
+                    margin: 0;
+                    margin-bottom: 15px;
+                }
```

Comment:
some of this styling seems general to Leaflet. It may impact other features, so I'd rather not have it here. Better to make it apply to this plugin only, or include it in the map styling. But for the latter that's possibly dangerous, since it may break other features.

Reply:
I do not know how to apply css only to a single plugin.

I am also not sure how much of a problem it is. I think Folium is typically run inside a separate iframe, so general styling will be limited to a single instance. I ran this `timeline` together with the `timestamped_geojson` in a single Jupyter notebook and there was no interference. 

The second css rule is limited to the "leaflet-timeline-controls" class, which can only be a problem if someone decides to reuse that class.

For the first css rule: the `TimelineSlider` will take the bottom left controller position, so no other controls can be positioned there for this map. To make the slider take the full width, which looks much nicer, this styling is required. This will impact other controls that take the bottom left, but I think in practice it will not be a problem.